lib/spack/spack/build_systems/python.py

```diff
@@ -171,61 +171,77 @@ def setup_py(self, *args, **kwargs):
 
     # Standard commands
 
+    def setup_args(self, spec, prefix):
+        '''Arguments to pass to build before the target command'''
+        return []
+
     def build(self, spec, prefix):
         """Build everything needed to install."""
-        args = self.build_args(spec, prefix)
+        args = (self.setup_args(spec, prefix),
+                + ['build']
+                + self.build_args(spec, prefix))
```

Comment:
If you instead call `setup_args` once in the `setup_py` function, you won't have to modify all of these other functions.

Reply:
```setup_py``` doesn't have access to spec and prefix, is that a problem?

Reply:
You should be able to get it from `self.spec`.

Reply:
You can get prefix from there too?

Reply:
Yep, `self.prefix`

Reply:
Cool!  Thanks for the patience.  Marking as resolved.